### PR TITLE
BLD: Remove CUDA requirements from recipe

### PR DESCRIPTION
**Pull request by carterbox**
Using a CUDA enabled version of PtychoNN is not strictly necessary. Persons using PtychoNN on apple silicon or whatever should be allowed to run Ptychodus will all the features even if PtychoNN is slower. If PtychoNN does require the CUDA version of pytorch, then it should export that requirement itself.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by carterbox** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,23 +10,16 @@ source:
...
 
     build:
       noarch: generic
-      ignore_run_exports_from:
-        - {{ compiler('cuda') }}
 
     requirements:
-      build:
-      # We don't need the compiler, just the __cuda virtual package
-        - {{ compiler('cuda') }}
```
Comment (carterbox) [inline recipe/meta.yaml:29]: Might need to keep these in order for import tests to pass.

**Comment by carterbox** [comment]
@stevehenke Regardless of if we remove this explicit requirement, Hanna still cannot use the `ptychodus-all` package on a Mac because *tike* requires CUDA.

**Comment by carterbox** [comment]
@stevehenke, we can also make a repodata patch to already published versions of ptychodus, if you think people want to run older versions with non-CUDA pytorch?

**Review by stevehenke — APPROVED** [review]
Thanks for figuring this out! I don't expect that there's much value in patching packages for older versions, but I could be wrong.

**Pull request merged**